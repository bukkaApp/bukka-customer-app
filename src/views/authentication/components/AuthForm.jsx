import React from 'react';

import PropTypes from 'prop-types';

import AuthButtonGroup from './AuthButtonGroup';
import AuthFooter from './AuthFooter';
import AuthInputGroup from './AuthInputGroup';
import TermsAndConditions from './TermsAndConditions';

import './authform.scss';

const ErrorMessage = ({ message }) => {
  if (message) {
    return (
      <div className="text-danger help-block text-center my-4">{message}</div>
    );
  }
  return null;
};

const preventDefault = event => event.preventDefault();

const AuthForm = ({
  handleLinkOptions,
  handleChange,
  handleSubmit,
  domStructure,
  validationErrors,
  isFormCompleted,
  errorMessage,
  title,
  slideToNextInput,
  userEmail,
  inputData,
}) => (
  <form
    className="pb-3 form-auth"
    name="form"
    onSubmit={!isFormCompleted ? preventDefault : handleSubmit}
  >
    <AuthInputGroup
      slideToNextInput={slideToNextInput}
      title={title}
      userEmail={userEmail}
      handleChange={handleChange}
      domStructure={domStructure}
      inputData={inputData}
      handleLinkOptions={handleLinkOptions}
      validationErrors={validationErrors}
    />
    <ErrorMessage message={errorMessage} />
    <TermsAndConditions
      handleLinkOptions={handleLinkOptions}
      title={title}
    />
    <AuthButtonGroup
      isFormCompleted={isFormCompleted}
      title={title}
    />
    <AuthFooter title={title} />
  </form>
);

export default AuthForm;

AuthForm.defaultProps = {
  domStructure: {},
  isFormCompleted: false,
  title: '',
  validationErrors: {},
  errorMessage: '',
  slideToNextInput: false,
  userEmail: '',
  handleLinkOptions: () => {}
};

AuthForm.propTypes = {
  handleLinkOptions: PropTypes.func,
  slideToNextInput: PropTypes.bool,
  userEmail: PropTypes.string,
  handleChange: PropTypes.func.isRequired,
  handleSubmit: PropTypes.func.isRequired,
  domStructure: PropTypes.arrayOf(PropTypes.objectOf(PropTypes.string)),
  validationErrors: PropTypes.objectOf(PropTypes.string),
  isFormCompleted: PropTypes.bool,
  title: PropTypes.string,
  errorMessage: PropTypes.string,
  inputData: PropTypes.objectOf(PropTypes.oneOfType([
    PropTypes.string, PropTypes.number])).isRequired,
};

ErrorMessage.defaultProps = {
  message: '',
};

ErrorMessage.propTypes = {
  message: PropTypes.string,
};
