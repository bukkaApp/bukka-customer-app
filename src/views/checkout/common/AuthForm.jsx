import React from 'react';

import Input from 'Components/input/InputField';
import Address from './Address';

const AuthForm = ({
  inputData,
  autoComplete,// eslint-disable-line
  inputField,
  handleChange,
  errors
}) =>
  inputField.map(propData => (
    <div className={propData.containerClassNames} key={propData.name}>
      <label
        htmlFor={propData.id}
        className={inputData[propData.name] ? 'fly-over m-0' : 'no-label'}
      >
        {propData.placeholderText}
      </label>
      {propData.name !== 'address' && <Input
        inputElement={{ autoComplete: 'off' }}
        type={propData.type}
        name={propData.name}
        handleChange={handleChange}
        classNames={propData.classNames}
        value={inputData[propData.name]}
        placeholderText={propData.placeholderText}
        id={propData.id}
        handleFocus={() => {}}
      />}
      {propData.name === 'address' && (
        <Address propData={propData} handleInputChange={handleChange} />
      )}
      <span className="text-danger font-size-11">{errors[propData.name]}</span>
    </div>
  ));

export default AuthForm;
