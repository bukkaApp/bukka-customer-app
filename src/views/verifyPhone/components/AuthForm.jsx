import React, { useEffect } from 'react';
import PropTypes from 'prop-types';
import { Spinner } from 'react-bootstrap';
import Button from '../../../components/button/Button';
import InputField from '../../../components/input/InputField';
import AuthService from '../../../shared/authServices';

const AuthForm = ({
  handleSubmit,
  handleChange,
  sent,
  errors,
  errorMessage,
  stopPhoneSpinner,
  inputData,
}) => {
  useEffect(() => {
    if (errorMessage !== '') {
      stopPhoneSpinner();
    }
  }, [errorMessage]);

  return (
    <>
      <h3>Add your phone number</h3>
      <div className="sub-text">
        <div className="p">Please verify your phone number</div>
        <div className="p">for your account</div>
        <div className="p bold">{AuthService.getEmail()}</div>
      </div>
      <div className="input-section">
        <div className="form-group">
          <div className="space">
            <div className="d-flex">
              <div className="p mt-2 prefix">+234</div>
              <InputField
                name="contactMobile"
                classNames="default-input"
                type="text"
                placeholderText="phone"
                handleChange={handleChange}
                value={inputData.contactMobile}
                error={errors.contactMobile}
                handleFocus={() => {}}
              />
            </div>
            <small className="text-danger error-margin-left">
              {errors.contactMobile}
            </small>
          </div>
        </div>
        <Button
          type="button"
          handleClick={handleSubmit}
          classNames="primary-button w-100"
        >
          <>
            {sent ? (
              <Spinner
                as="span"
                aria-hidden="true"
                animation="border"
                variant="light"
              />
            ) : (
              'Submit'
            )}
          </>
        </Button>
      </div>
    </>
  );
};

AuthForm.propTypes = {
  handleSubmit: PropTypes.func.isRequired,
  handleChange: PropTypes.func.isRequired,
  sent: PropTypes.bool.isRequired,
  inputData: PropTypes.objectOf(PropTypes.oneOfType([PropTypes.string, PropTypes.number])).isRequired,
  errors: PropTypes.objectOf(PropTypes.string).isRequired,
  errorMessage: PropTypes.string.isRequired,
  stopPhoneSpinner: PropTypes.func.isRequired,
};

export default AuthForm;
