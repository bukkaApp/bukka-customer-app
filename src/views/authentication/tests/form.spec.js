import React from 'react';
import AuthForm from 'src/components/AuthForm';
import Data from './__mocks__/register';


describe('Auth Form component', () => {
  const props = {
    handleChange: jest.fn(),
    handleSubmit: jest.fn(),
    inputData: Data,
    isFormCompleted: false,
    type: 'signin',
    domStructure: Data.domStructureMock
  };
  const wrapper = shallow(<AuthForm {...props} />);

  it('renders properly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('traverse the form component', () => {
    expect(wrapper.find('form').first()).toBeTruthy();
    expect(wrapper.find('.divider-box')).toBeTruthy();
  });
});
