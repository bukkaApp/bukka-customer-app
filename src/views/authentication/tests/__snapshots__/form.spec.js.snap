// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Auth Form component renders properly 1`] = `ShallowWrapper {}`;
